### PR TITLE
Adding user deactivate key and cron job to run it.

### LINKED ISSUE #107: Add 'deactivate account on date' field to user accounts
In order to make it easier for Admin users to manage users, add a 'deactivate account on date' field (or some similar name like `deactivateOn`). The `deactivateOn` field should be a date that the account should automatically be marked as inactive. Set up an automated job to deactivate accounts on a daily basis.

### PR DESCRIPTION
**Pull request by shailee-m**
closes #107

### CODE REVIEW

File: app/.meteor/packages [hunk 1]
```diff
@@ -52,3 +52,4 @@ akryum:vue-blaze-template
 underscore
 vuejs:blaze-integration
 akryum:vue-sass
+littledata:synced-cron
```
Comment (brylie) [inline app/.meteor/packages]: Just make sure this package is somewhat maintained, as a lot of Meteor specific packages are being abandoned. Where is the official repository for this project? I can't find a repo with the 1.5.1 release.

**Comment by brylie** [comment]
Looking good so far. Lets just be cautious about new dependencies, as they may cause problems when upgrading the project, due to version constraints :-)

File: app/.meteor/packages (hunk 1 above)
Comment (shailee-m) [reply app/.meteor/packages]: @brylie Here is the repository: https://github.com/percolatestudio/meteor-synced-cron
It has MIT liscensing and over 70k installs showing on atmosphere and did not have a lot of open issues. Is there anything else I should look for?

Comment (brylie) [reply app/.meteor/packages]: Well, just make sure there is a *recent* release, i.e. that it is not abandoned. It should be OK, but I have just had trouble with some parts of the Meteor ecosystem getting abandoned, like Blaze.

Comment (shailee-m) [reply app/.meteor/packages]: Alright will take care of it.

File: app/server/crons/deactivateUsers.js [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
-  },
-  job: function() {
+import { CronJob } from 'cron';
+
+new CronJob(
+  '0 0 12 * * *',
+  Meteor.bindEnvironment(function(err, data) {
     /* Get all users for whom the deactivation date is set */
     const users = Meteor.users
       .find({
```
Comment (brylie) [inline app/server/crons/deactivateUsers.js]: This query should probably filter out users who are already deactivated, i.e. where `isActive` is already `false`.

File: app/server/crons/deactivateUsers.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+new CronJob(
+  '0 0 12 * * *',
+  Meteor.bindEnvironment(function(err, data) {
+    /* Get all users for whom the deactivation date is set */
+    const users = Meteor.users
+      .find({
+        $and: [
+          { deactivateOn: { $lte: new Date() } },
+          { deactivateOn: { $gt: new Date(0) } },
```
Comment (brylie) [inline app/server/crons/deactivateUsers.js]: Why do we need this `$gt` clause? shouldn't the `$lte` be sufficient? We can also require that the `deactivatedOn` is set, with `exists: true`, or something similar.

Comment (shailee-m) [reply app/server/crons/deactivateUsers.js]: We need to set the deactivated on records for mongo to be able to find that key in the document. So I had set all the deactivateOn fields to timestamp=0

File: app/server/crons/deactivateUsers.js (hunk 2 above)
Comment (shailee-m) [reply app/server/crons/deactivateUsers.js]: Okay I think I can handle the update in only one query itself. Will change that.

File: app/server/crons/deactivateUsers.js (hunk 3 above)
Comment (brylie) [reply app/server/crons/deactivateUsers.js]: Would `{deactivatedOn: {exists: true}}` have the same result?

Comment (shailee-m) [reply app/server/crons/deactivateUsers.js]: No it wouldn't. I realised setting deactivateOn by default was unnecessary so I removed it.

File: app/lib/accounts.js [hunk 4]
```diff
@@ -44,7 +43,7 @@ Meteor.startup(function() {
     var { isActive } = attempt.user;
 
     /* If the user is inactive throw an error */
-    if (!isActive) {
+    if (isActive === false) {
       attempt.allowed = false;
       throw new Meteor.Error(403, 'User account is inactive!');
```
Comment (brylie) [inline app/lib/accounts.js]: Lets not shout at people. We can just say "User account is inactive."

**Comment by brylie** [comment]
@PayalChoksey17 can you do a QA round on this pull request?

**Comment by PayalChoksey17** [comment]
@shailee-m @brylie  Can you help me with this feature where it is implemented with a screenshot.

**Comment by shailee-m** [comment]
> @shailee-m @brylie Can you help me with this feature where it is implemented with a screenshot.

@PayalChoksey17  I have added the deactivation date in settings>user > users' table (Inside the edit user popup).

After the selected deactivation date has passed, the user should not be able to login.

Currently the job is set to run everyday at 12PM. So for testing purpose you can change the frequency here: `app/server/crons/deactivateUsers.js `. Set this  '0 0 12 * * *' to  '0 2 * * * *' to run every 2 minutes.

To set any other frequency, use the following links: 
1. https://www.freeformatter.com/cron-expression-generator-quartz.html#
2. https://cronjob.xyz/

**Comment by PayalChoksey17** [comment]
thanks @shailee-m Shailee Mehta

**Comment by PayalChoksey17** [comment]
@shailee-m  Will the Admin also have the rights to deactivate his own account ? I see that when I login as admin, the Admin can also set the deactivate date for his own account. 
Screenshot_1

**Comment by shailee-m** [comment]
> @shailee-m Will the Admin also have the rights to deactivate his own account ? I see that when I login as admin, the Admin can also set the deactivate date for his own account.
> Screenshot_1

Untill now we were allowing revoking admin (basically all actions same as other users) but I think to remove it will be better.

**Pull request closed without merging**